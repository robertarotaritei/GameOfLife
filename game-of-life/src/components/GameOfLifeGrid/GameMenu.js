import React from 'react';
import { ButtonToolbar } from 'react-bootstrap';
import PlayArrowIcon from '@material-ui/icons/PlayArrow';
import { IconButton } from '@material-ui/core';
import PauseIcon from '@material-ui/icons/Pause';
import RestoreIcon from '@material-ui/icons/Restore';
import ClearIcon from '@material-ui/icons/Clear';
import AcUnitIcon from '@material-ui/icons/AcUnit';
import FastForwardIcon from '@material-ui/icons/FastForward';
import SpaIcon from '@material-ui/icons/Spa';
import SaveIcon from '@material-ui/icons/Save';
import SpeedIcon from '@material-ui/icons/Speed';

class GameMenu extends React.Component {
	constructor() {
		super();

		this.state = {
			speed: false
		}
	}

	changeSpeed = (e) => {
		e.preventDefault();
		this.setState({ speed: !this.state.speed });
	}

	render() {
		return (
			<div>
				<ButtonToolbar>
					{this.props.playState === 'play' ? (
						<IconButton data-testid='play' onClick={this.props.playButton} >
							<PlayArrowIcon />
						</IconButton>
					) : (
							<IconButton data-testid='pause' onClick={this.props.pauseButton}>
								<PauseIcon />
							</IconButton>
						)
					}
					<IconButton data-testid='slow' onClick={this.changeSpeed}>
						<SpeedIcon />
					</IconButton>
					{this.state.speed ? (
						<IconButton data-testid='slow' onClick={this.props.slow}  size='small' style={{backgroundColor: '#93b498'}} >
							<AcUnitIcon />
						</IconButton>
					) : null
					}
					{this.state.speed ? (
						<IconButton data-testid='fast' onClick={this.props.fast} size='small' style={{backgroundColor: '#93b498'}} >
							<FastForwardIcon />
						</IconButton>
					) : null
					}
					<IconButton data-testid='back' onClick={this.props.stop}>
						<RestoreIcon />
					</IconButton>
					<IconButton data-testid='reset' onClick={this.props.clear}>
						<ClearIcon />
					</IconButton>
					{this.props.history ?
						null : (
							<IconButton data-testid='seed' onClick={this.props.seed}>
								<SpaIcon />
							</IconButton>
						)
					}
					{this.props.history ?
						null : (
							<IconButton data-testid='save' onClick={this.props.save}>
								<SaveIcon />
							</IconButton>
						)
					}
				</ButtonToolbar>
			</div>
		)
	}
}

export default GameMenu;