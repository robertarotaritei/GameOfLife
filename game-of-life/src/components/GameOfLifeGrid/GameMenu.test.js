import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import GameMenu from './GameMenu';
import { shallow } from "enzyme";

test("GameMenu renders without crashing", () => {
  shallow(<GameMenu />);
});

test('Render GameMenu', () => {
  const { getByText } = render(<GameMenu playState={'play'}/>);
  const linkElement = getByText(/Play/i);
  expect(linkElement).toBeInTheDocument();
});

it('GameMenu renders correctly', () => {
  const tree = renderer
    .create(<GameMenu />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render GameMenu', () => {
  const { getByText } = render(<GameMenu playState={'resume'}/>);
  const linkElement = getByText(/Resume/i);
  expect(linkElement).toBeInTheDocument();
});