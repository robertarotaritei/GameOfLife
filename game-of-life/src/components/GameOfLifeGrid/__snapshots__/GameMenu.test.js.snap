// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMenu renders correctly 1`] = `
<div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="btn-default"
    >
      Resume
    </button>
    <button
      className="btn-default"
    >
      Pause
    </button>
    <button
      className="btn-default"
    >
      Back to First Generation
    </button>
    <button
      className="btn-default"
    >
      Reset
    </button>
    <button
      className="btn-default"
    >
      Slow
    </button>
    <button
      className="btn-default"
    >
      Fast
    </button>
    <div>
      <button
        className="btn-default"
      >
        Seed
      </button>
      <button
        className="btn-default"
      >
        Save First Generation
      </button>
    </div>
  </div>
</div>
`;
